core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -171,7 +172,10 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         builders.put(Names.INDEX, new FixedExecutorBuilder(settings, Names.INDEX, availableProcessors, 200));
         builders.put(Names.BULK, new FixedExecutorBuilder(settings, Names.BULK, availableProcessors, 200)); // now that we reuse bulk for index/delete ops
         builders.put(Names.GET, new FixedExecutorBuilder(settings, Names.GET, availableProcessors, 1000));
-        builders.put(Names.SEARCH, new FixedExecutorBuilder(settings, Names.SEARCH, searchThreadPoolSize(availableProcessors), 1000));
+        // The search threadpool's queue will automatically be adjusted. It starts initially at 1000
+        // with a minimum of 10 and maximum of 2000. The default frame size is 2000 operations
```

Comment:
I appreciate code comments, but I think this one can be dropped. It doesn't communicate anything that the code doesn't already show and it's at risk of becoming inconsistent with the code over time as it happens so frequently that a change is made to the code but comments aren't updated. What I would support if you think this comment is essential is:

```java
builders.put(
                Names.SEARCH,
                new AutoQueueAdjustingExecutorBuilder(
                        settings,
                        Names.SEARCH,
                        searchThreadPoolSize(availableProcessors),
                        1000 /* initial queue size */,
                        10 /* min queue size */,
                        2000 /* max queue size */,
                        2000 /* frame size */));
```

but even this I think is unnecessary.